fast-logger/System/Log/FastLogger/Logger.hs

```diff
@@ -21,20 +20,19 @@ import System.Log.FastLogger.LogStr
 
 ----------------------------------------------------------------
 
-data Logger = Logger !BufSize (MVar Buffer) (IORef LogStr)
+newtype Logger = Logger (IORef LogStr)
 
 ----------------------------------------------------------------
 
-newLogger :: BufSize -> IO Logger
-newLogger size = Logger size <$> (getBuffer size >>= newMVar)
-                             <*> newIORef mempty
+newLogger :: IO Logger
+newLogger = Logger <$> newIORef mempty
 
 ----------------------------------------------------------------
 
-pushLog :: IORef FD -> Logger -> LogStr -> IO ()
-pushLog fdref logger@(Logger size mbuf ref) nlogmsg@(LogStr nlen nbuilder)
+pushLog :: IORef FD -> BufSize -> MVar Buffer -> Logger -> LogStr -> IO ()
+pushLog fdref size mbuf logger@(Logger ref) nlogmsg@(LogStr nlen nbuilder)
```

Comment:
Writing is protected by the same `MVar` when message size is longer than buffer size. OK.